AppDB/appscale/datastore/datastore_distributed.py

```diff
@@ -1002,10 +1003,21 @@ def __get_start_key(self, prefix, prop_name, order, last_result):
       plist = ent.property_list()
 
     val = None
-    for p in plist:
-      if p.name() == prop_name:
-        val = str(encode_index_pb(p.value()))
-        break
+
+    # If the entity provided by the cursor has multiple values for the given
+    # property and the query has information about what the value should be,
+    # use the value from the query.
+    if (query is not None and query.filter_size() == 1 and
+        query.filter(0).op() == datastore_pb.Query_Filter.EQUAL):
```

Comment:
We will need to do more work to behave properly in those scenarios. The value of the cursor entity can be ambiguous when the filter is not EQUAL.

The GAE docs mention that using a cursor with an inequality filter for a property with multiple values gives unexpected results: https://cloud.google.com/appengine/docs/standard/python/datastore/query-cursors#limitations_of_cursors, but we need to make sure we do the same thing (which probably does not involve an infinite loop).